frontend/javascripts/oxalis/model/sagas/skeletontracing_saga.ts

```diff
@@ -265,20 +265,35 @@ function* getAgglomerateSkeletonTracing(
     const parsedTracing = parseProtoTracing(nmlProtoBuffer, "skeleton");
 
     if (!("trees" in parsedTracing)) {
-      // This check is only for flow to realize that we have a skeleton tracing
+      // This check is only for typescript to realize that we have a skeleton tracing
       // on our hands.
       throw new Error("Skeleton tracing doesn't contain trees");
     }
 
+    if (parsedTracing.trees.length !== 1) {
+      throw new Error(
+        `Agglomerate skeleton response does not contain exactly one tree, but ${parsedTracing.trees.length} instead.`,
+      );
+    }
+
+    // Make sure the tree is named as expected
+    parsedTracing.trees[0].name = getTreeNameForAgglomerateSkeleton(agglomerateId, mappingName);
```

Comment:
Yes exactly, I discussed this with @fm3 internally. This solution fixes two issues at once.
For one, the mentioned bug where agglomerate skeletons didn't have the expected name sometimes if an editable mapping was active.
Additionally, the frontend implicitly assumed that the agglomerate skeleton naming by the backend was deterministic - if it ever were to be changed it would have had to be adapted in both back- and frontend which is a source of potential bugs.

This way, no more assumption by the frontend is needed since the `getTreeNameForAgglomerateSkeleton` function is the definitive source of truth for agglomerate skeleton names.